src/library/trackmodel.h

```diff
@@ -144,7 +146,20 @@ class TrackModel {
     virtual void select() {
     }
 
+    virtual int getPosition(){
+      return position;
+    }
+
+    virtual void setPosition(int newPosition){
+      if (newPosition <= 100 && newPosition >= 0 && abs(this->position-newPosition) < 10){
```

Comment:
Why is this check required? The constants should be used by name like 
const int kFancyPositionChangeLimit = 10;